src/server.cpp

```diff
@@ -2373,20 +2375,32 @@ void Server::fillMediaCache()
 	}
 }
 
-void Server::sendMediaAnnouncement(u16 peer_id)
+void Server::sendMediaAnnouncement(u16 peer_id, const std::string &lang_code)
 {
 	DSTACK(FUNCTION_NAME);
 
 	verbosestream << "Server: Announcing files to id(" << peer_id << ")"
 		<< std::endl;
 
 	// Make packet
-	std::ostringstream os(std::ios_base::binary);
-
 	NetworkPacket pkt(TOCLIENT_ANNOUNCE_MEDIA, 0, peer_id);
-	pkt << (u16) m_media.size();
+
+	u16 media_sent = 0;
+	std::string lang_suffix;
+	lang_suffix.append(".").append(lang_code).append(".tr");
```

Comment:
This was the original code; @nerzhul suggested using append that has better performance instead.